awswrangler/lakeformation/_read.py

```diff
@@ -0,0 +1,192 @@
+
+    Note
+    ----
+    `chunked` argument (memory-friendly):
+    If set to `True`, return an Iterable of DataFrames instead of a regular DataFrame.
+
+    Parameters
+    ----------
+    sql : str
+        partiQL query.
+    database : str
+        AWS Glue database name
+    transaction_id : str, optional
+        The ID of the transaction at which to read the table contents.
+        Cannot be specified alongside query_as_of_time
+    query_as_of_time : str, optional
+        The time as of when to read the table contents. Must be a valid Unix epoch timestamp.
+        Cannot be specified alongside transaction_id
+    catalog_id : str, optional
+        The ID of the Data Catalog from which to retrieve Databases.
+        If none is provided, the AWS account ID is used by default.
+    chunked : bool, optional
+        If `True`, Wrangler returns an Iterable of DataFrames with no guarantee of chunksize.
+    use_threads : bool
+        True to enable concurrent requests, False to disable multiple threads.
+        When enabled, os.cpu_count() is used as the max number of threads.
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session is used if boto3_session receives None.
+    params: Dict[str, any], optional
+        Dict of parameters used to format the partiQL query. Only named parameters are supported.
+        The dict must contain the information in the form {"name": "value"} and the SQL query must contain
+        `:name`.
+
+    Returns
+    -------
+    Union[pd.DataFrame, Iterator[pd.DataFrame]]
+        Pandas DataFrame or Generator of Pandas DataFrames if chunked is passed.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> df = wr.lakeformation.read_sql_query(
+    ...     sql="SELECT * FROM my_table;",
+    ...     database="my_db",
+    ...     catalog_id="111111111111"
+    ... )
+
+    >>> import awswrangler as wr
+    >>> df = wr.lakeformation.read_sql_query(
+    ...     sql="SELECT * FROM my_table LIMIT 10;",
+    ...     database="my_db",
+    ...     transaction_id="1b62811fa3e02c4e5fdbaa642b752030379c4a8a70da1f8732ce6ccca47afdc9"
+    ... )
+
+    >>> import awswrangler as wr
+    >>> df = wr.lakeformation.read_sql_query(
+    ...     sql="SELECT * FROM my_table WHERE name=:name;",
+    ...     database="my_db",
+    ...     query_as_of_time="1611142914",
+    ...     params={"name": "'filtered_name'"}
+    ... )
+
+    """
+    if transaction_id is not None and query_as_of_time is not None:
+        raise exceptions.InvalidArgumentCombination(
+            "Please pass only one of `transaction_id` or `query_as_of_time`, not both"
+        )
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+    if params is None:
+        params = {}
+    for key, value in params.items():
+        sql = sql.replace(f":{key}", str(value))
```

Comment:
Question:
I am not sure I understand why the [Athena logic for this](https://github.com/awslabs/aws-data-wrangler/blob/54a266c860a1a9a6d4ec31aaaf6f22ccf92c1b5c/awswrangler/athena/_read.py#L782) references a `;` in the replace method.

`;` should only appears at the end of the SQL statement. Putting it in the `replace` prevents multiple replacements if there is more than one key in the `params` dictionary no?

Reply:
Multiple params should still be possible. The `;` is just a delimiter where the substitution should end. Example:
```python
sql = "SELECT * FROM mytable WHERE customerID=:customer_id; AND city=:city;"
params = {"customer_id": 123, "city": "Munich"}
for key, value in params.items():
        sql = sql.replace(f":{key};", str(value))
print(sql)
>>> 'SELECT * FROM mytable WHERE customerID=123 AND city=Munich'
```

Also if you want to have a semicolon at the end of the query you have to put a double semicolon at the end:
```python
sql = "SELECT * FROM mytable WHERE customerID=:customer_id; AND city=:city;;"
params = {"customer_id": 123, "city": "Munich"}
for key, value in params.items():
        sql = sql.replace(f":{key};", str(value))
print(sql)
>>> 'SELECT * FROM mytable WHERE customerID=123 AND city=Munich;'
```

Reply:
Ah ok I see now, thanks @maxispeicher. But then why have the `;` at all? I feel like it's confusing SQL users since it's supposed to mean the end of a SQL statement. Wouldn't `sql = sql.replace(f":{key}", str(value))` work just as well?

Reply:
From my side it was just to be specific about where the substitution will take place, but I'm also open to changing this. However then also the documentation and the examples would need to be adapted.